DJHelper/DJHelper/Helpers/ShiftableViewController.swift

```diff
@@ -0,0 +1,131 @@
+//
+//  ShiftableViewController.swift
+//  DJHelper
+//
+//  Created by Craig Swanson on 6/10/20.
+//  Copyright © 2020 craigswanson. All rights reserved.
+//
+
+import UIKit
+
+class ShiftableViewController: UIViewController, UITextFieldDelegate, UITextViewDelegate, UIGestureRecognizerDelegate {
+
+    var currentYShiftForKeyboard: CGFloat = 0
+
+    var textFieldBeingEdited: UITextField?
+    var textViewBeingEdited: UITextView?
+
+    var keyboardDismissTapGestureRecognizer: UITapGestureRecognizer!
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        setupKeyboardDismissTapGestureRecognizer()
+
+        NotificationCenter.default.addObserver(self,
+                                               selector: #selector(keyboardWillShow(notification:)),
+                                               name: UIResponder.keyboardWillShowNotification,
+                                               object: nil)
+        NotificationCenter.default.addObserver(self,
+                                               selector: #selector(keyboardWillHide(notification:)),
+                                               name: UIResponder.keyboardWillHideNotification,
+                                               object: nil)
+    }
+
+    @objc func stopEditingTextInput() {
+        if let textField = self.textFieldBeingEdited {
+
+            textField.resignFirstResponder()
+
+            self.textFieldBeingEdited = nil
+            self.textViewBeingEdited = nil
+        } else if let textView = self.textViewBeingEdited {
+
+            textView.resignFirstResponder()
+
+            self.textFieldBeingEdited = nil
+            self.textViewBeingEdited = nil
+        }
+
+        guard keyboardDismissTapGestureRecognizer.isEnabled else { return }
+
+        keyboardDismissTapGestureRecognizer.isEnabled = false
+    }
+
+    func textFieldDidBeginEditing(_ textField: UITextField) {
+        textFieldBeingEdited = textField
+    }
+
+    func textViewShouldBeginEditing(_ textView: UITextView) -> Bool {
+        textViewBeingEdited = textView
+        return true
+    }
+
+    @objc func keyboardWillShow(notification: Notification) {
```

Comment:
If I understand correctly this is to move the keyboard to a certain position. Good work with this logic!!!